### ISSUE TITLE
Bug in tf.test.compute_gradient for tf.keras.layers.LeakyReLU

### ISSUE THREAD

**Issue by cheyennee** [tensorflow/tensorflow#62436]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

tf 2.14.0

### Custom code

Yes

### OS platform and distribution

windows colab

### Python version

3.10

### Current behavior?

The numerical gradient and theoretical gradient for `tf.keras.layers.LeakyReLU` is different, one is `0.2`, the other is `0.6`. 

### Standalone code to reproduce the issue

```shell
import tensorflow as tf

alpha = 0.2
__input___0_tensor = tf.random.uniform([2, 1, 1, 1], minval=0, maxval=0,
                                                    dtype=tf.float64)
__input___0 = tf.identity(
    __input___0_tensor)
LeakyReLU_class = tf.keras.layers.LeakyReLU(alpha=alpha, dtype=tf.float64)

layer = LeakyReLU_class
inputs = __input___0

r = LeakyReLU_class(inputs)
theoretical, numerical = tf.test.compute_gradient(LeakyReLU_class, [inputs])
print(theoretical)
print(numerical)
```

### Relevant log output

```shell
(array([[0.2, 0. ],
       [0. , 0.2]]),)
(array([[0.6, 0. ],
       [0. , 0.6]]),)
```

**Comment by SuryanarayanaY** [comment]
Hi @cheyennee ,

I have replicated the reported error and attached the gist [here](https://colab.sandbox.google.com/gist/SuryanarayanaY/adf56731c80d34c22f6bc9cb774bae76/62436.ipynb).Need to check for the difference in results.

**Comment by LakshmiKalaKadali** [comment]
Hi @cheyennee,

The difference between the numerical gradient and theoretical gradient is with input initialization. In the code, `__input___0_tensor = tf.random.uniform([2, 1, 1, 1], minval=0, maxval=0, dtype=tf.float64)`, the min and max values are set to '0' means it's a zero tensor. Please set either min or max values to non zero. Verified with TF 2.17.1 and latest stable version TF 2.18.0. It is working fine. Please refer to the [gist](https://colab.sandbox.google.com/gist/LakshmiKalaKadali/787799bfca046cf2264631641d9764c3/tf_62436.ipynb). 

Thank You
